CoreFramework/Reporter/HtmlReport.cs

```diff
@@ -0,0 +1,194 @@
+    }
+
+    public static ExtentTest GetParent()
+    {
+        return _extentTestSuite;
+    }
+
+
+    public static ExtentTest GetNode()
+    {
+        return _extentTestCase;
+    }
+
+    public static ExtentTest GetTest()
+    {
+        if (GetNode() == null)
+        {
+            return GetParent();
+        }
+        return GetNode();
+    }
+    #endregion
+
+    #region TestContext for Fail/Pass/Info/Skip
+    public static void Pass(string des)
+    {
+
+        GetTest().Pass(des);
+        TestContext.WriteLine(des);
+    }
+
+    public static void Pass (string des, string path)
+    {
+        GetTest().Pass(des).AddScreenCaptureFromPath(path);
+        TestContext.WriteLine(des);
+    }
+
+    public static void Fail(string des)
+    {
+        GetTest().Fail(des);
+        TestContext.WriteLine(des);
+    }
+
+    public static void Fail(string des, string path)
+    {
+
+        GetTest().Fail(des).AddScreenCaptureFromPath(path);
+        TestContext.WriteLine(des);
+    }
+
+    public static void Fail (string des, string ex, string path)
+    {
+        GetTest().Fail(des).Fail(ex).AddScreenCaptureFromPath(path);
+        TestContext.WriteLine(des);
+    }
+
+    public static void Info(string des)
+    {
+        GetTest().Info(des);
+        TestContext.WriteLine(des);
+    }
+    public static void Info(APIRequest request, APIResponse response)
+    {
+        GetTest().Info(MarkupHelperPlus.CreateAPIRequestLog(request, response));
+        TestContext.Progress.WriteLine();
+    }
+    public static void Warning (string des)
+    {
+        GetTest().Warning(des);
+        TestContext.WriteLine(des);
+    }
+    public static void Skip(string des)
+    {
+        GetTest().Skip(des);
+        TestContext.WriteLine(des);
+    }
+
+    #endregion
+
+    #region  Markup 
+    public static void MarkupPassJson(string json)
+    {
+      GetTest().Info(MarkupHelper.CreateCodeBlock(json, CodeLanguage.Json));
+    }
+    public static void MarkupTable(string[][] someInts)
+    {
+        GetTest().Info(MarkupHelper.CreateTable(someInts));
+    }
+    public static void MarkupPassLabel(string text)
+    {
+        GetTest().Pass(MarkupHelper.CreateLabel(text, ExtentColor.Green));
+    }
+    public static void MarkupFailLabel(string text)
+    {
+        GetTest().Fail(MarkupHelper.CreateLabel(text, ExtentColor.Red));
+    }
+    public static void MarkupWarningLabel(string text)
+    {
+        GetTest().Warning(MarkupHelper.CreateLabel(text, ExtentColor.Orange));
+    }
+    public static void MarkupSkipLabel(string text)
+    {
+        GetTest().Skip(MarkupHelper.CreateLabel(text, ExtentColor.Blue));
+    }
+    public static void MarkupXML(string code)
+    {
+        GetTest().Info(MarkupHelper.CreateCodeBlock(code, CodeLanguage.Xml));
+    }
+    #endregion 
+
+    #region  Log for API requests/response
+    public static void CreateAPIRequestLog_(APIRequest request, APIResponse response)
```

Comment:
Remove meaningless syntax or word from method name such as _

Reply:
Done for other files as well